external/fv3fit/fv3fit/data/tfdataset.py

```diff
@@ -3,11 +3,57 @@
 import tensorflow as tf
 from .base import TFDatasetLoader, register_tfdataset_loader
 import dacite
+from ..tfdataset import iterable_to_tfdataset
+import tempfile
+import xarray as xr
+import numpy as np
+from fv3fit._shared.stacking import stack
 
 
 @dataclasses.dataclass
 class VariableConfig:
-    unstacked_dims: Sequence[str]
+    """
+    Configuration for a variable to retrieve from a dataset.
+
+    Attributes:
+        times: time indices to use, must be one of "window" or "start", which
+            will use either the full window or just the initial time of that window
+    """
+
+    times: str = "window"
+
+    def __post_init__(self):
+        if self.times not in ("window", "start"):
+            raise TypeError("times must be one of 'window' or 'start'")
```

Comment:
An issue is we want to type hint this as `Literal['window', 'start']` but can only type hint it as str. I often use ValueError in these situations but feel bad about it, because classes almost always raise TypeError if you give them an invalid initialization argument.